kernel/common/file.rb

```diff
@@ -1195,6 +1195,21 @@ def self.sticky?(file_name)
     return false
   end
 
+  def self.mkfifo(file_name, mode = 0666)
+    raise NotImplementedError, "mkfifo is not implemented on this platform" unless Rubinius::HAVE_MKFIFO
```

Comment:
MRI has a special C function that is put in the method table when the method is not implemented on that platform. We do this for FFI, but we don't have a generic way to do it yet.

For now, this is fine. We already do it in a few places and if someone is doing mkfifo in a tight loop, they probably need to refactor something.